src/components/scenes/SendScene2.tsx

```diff
@@ -0,0 +1,924 @@
+          isCameraOpen={openCamera}
+          fioToAddress={fioAddress}
+          ref={() => {}}
+        />
+      )
+    }
+
+    return null
+  }
+
+  const handleAmountsChanged = (spendTarget: EdgeSpendTarget) => (amounts: ExchangedFlipInputAmounts) => {
+    const { nativeAmount, fieldChanged: newField } = amounts
+    spendTarget.nativeAmount = nativeAmount === '' ? undefined : nativeAmount
+
+    // This works since the spendTarget object is guaranteed to be inside
+    // the spendInfo object
+    setSpendInfo({ ...spendInfo })
+    setMaxSpendSetter(0)
+    setFieldChanged(newField)
+  }
+
+  const handleSetMax = (index: number) => () => {
+    setMaxSpendSetter(index)
+  }
+
+  const handleFeesChange = useHandler(() => {
+    if (coreWallet == null) return
+
+    navigation.navigate('changeMiningFee2', {
+      spendInfo,
+      maxSpendSet: maxSpendSetter > 0,
+      wallet: coreWallet,
+      onSubmit: (networkFeeOption, customNetworkFee) => {
+        setSpendInfo({ ...spendInfo, networkFeeOption, customNetworkFee })
+        setMaxSpendSetter(0)
+        setPinValue(undefined)
+      }
+    })
+  })
+
+  const handleFlipInputModal = (index: number, spendTarget: EdgeSpendTarget) => () => {
+    Airship.show<FlipInputModalResult>(bridge => (
+      <FlipInputModal2
+        ref={flipInputModalRef}
+        bridge={bridge}
+        startNativeAmount={spendTarget.nativeAmount}
+        forceField={fieldChanged}
+        onAmountsChanged={handleAmountsChanged(spendTarget)}
+        onMaxSet={handleSetMax(index + 1)}
+        onFeesChange={handleFeesChange}
+        wallet={coreWallet}
+        tokenId={tokenId}
+        feeNativeAmount={feeNativeAmount}
+      />
+    ))
+      .then(async () => {
+        if (error == null) return
+        console.log(error)
+        const insufficientFunds = asMaybeInsufficientFundsError(error)
+        if (insufficientFunds != null && insufficientFunds.currencyCode != null && spendInfo.currencyCode !== insufficientFunds.currencyCode) {
+          const { currencyCode, networkFee = '' } = insufficientFunds
+          const multiplier = cryptoDisplayDenomination.multiplier
+          const amountString = roundedFee(networkFee, 2, multiplier)
+          const result = await Airship.show<'buy' | 'exchange' | 'cancel' | undefined>(bridge => (
+            <ButtonsModal
+              bridge={bridge}
+              title={s.strings.buy_crypto_modal_title}
+              message={`${amountString}${sprintf(s.strings.buy_parent_crypto_modal_message, currencyCode)}`}
+              buttons={{
+                buy: { label: sprintf(s.strings.buy_crypto_modal_buy_action, currencyCode) },
+                exchange: { label: s.strings.buy_crypto_modal_exchange, type: 'primary' },
+                cancel: { label: s.strings.buy_crypto_decline }
+              }}
+            />
+          ))
+          switch (result) {
+            case 'buy':
+              Actions.jump('pluginListBuy', { direction: 'buy' })
```

Comment:
Use `navigation`, since this is a brand-new scene.

Reply:
will do